model/avatar.go

```diff
@@ -41,6 +56,26 @@ func encodeToPNG(img image.Image) (raw []byte, err error) {
 	return
 }
 
+func decodeImage(raw []byte) (img image.Image, err error) {
+	format := http.DetectContentType(raw)
+	reader := bytes.NewReader(raw)
+
+	switch format {
+	case "image/gif":
+		img, err = gif.Decode(reader)
+	case "image/jpeg":
+		img, err = jpeg.Decode(reader)
+	case "image/png":
+		img, err = png.Decode(reader)
+	}
```

Comment:
What happens if I upload a `.docx`?

Maybe add default to switch which creates an error?
Like:
```go
switch format {
case "image/gif":
...
...
...
default:
          err = fmt.Errorf("unsupported format: '%s'", format")
```